tephra-hbase-compat-1.1-base/src/main/java/org/apache/tephra/hbase/SecondaryIndexTable.java

```diff
@@ -177,11 +173,23 @@ public void close() throws IOException {
     } catch (IOException e) {
       try {
         secondaryIndexTable.close();
+        conn.close();
```

Comment:
The ``finalize()`` would have helped in this scenario. Given that we don't want to use it as a standard, definitely we can go with either approach. Will make another commit with the changes.